src/mlpack/methods/ann/ffn.hpp

```diff
@@ -50,7 +50,9 @@ namespace mlpack {
 template<
     typename OutputLayerType = NegativeLogLikelihood,
     typename InitializationRuleType = RandomInitialization,
-    typename MatType = arma::mat>
+    typename MatType = arma::mat, 
+    typename NetworkType = MultiLayer<MatType> 
```

Comment:
Is there a reason we need to add this parameter?  I don't understand why it's necessary.